src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/OverseasEntityDueDiligenceValidator.java

```diff
@@ -0,0 +1,102 @@
+import uk.gov.companieshouse.overseasentitiesapi.validation.utils.UtilsValidators;
+import uk.gov.companieshouse.service.rest.err.Errors;
+
+import java.time.LocalDate;
+import java.util.Objects;
+
+import static uk.gov.companieshouse.overseasentitiesapi.validation.utils.ValidationUtils.getQualifiedFieldName;
+
+@Component
+public class OverseasEntityDueDiligenceValidator {
+
+    public static final int MAX_LENGTH = 256;
+    private final AddressDtoValidator addressDtoValidator;
+
+    @Autowired
+    public OverseasEntityDueDiligenceValidator(AddressDtoValidator addressDtoValidator) {
+        this.addressDtoValidator = addressDtoValidator;
+    }
+
+    public Errors validate(OverseasEntityDueDiligenceDto overseasEntityDueDiligenceDto, Errors errors, String loggingContext) {
+        validateIdentityDate(overseasEntityDueDiligenceDto.getIdentityDate(), errors, loggingContext);
+        validateName(overseasEntityDueDiligenceDto.getName(), errors, loggingContext);
+        validateAddress(overseasEntityDueDiligenceDto.getAddress(), errors, loggingContext);
+        validateEmail(overseasEntityDueDiligenceDto.getEmail(), errors, loggingContext);
+        if (Objects.nonNull(overseasEntityDueDiligenceDto.getSupervisoryName())) {
+            validateSupervisoryName(overseasEntityDueDiligenceDto.getSupervisoryName(), errors, loggingContext);
+        }
+        if (Objects.nonNull(overseasEntityDueDiligenceDto.getAmlNumber())) {
+            validateAmlNumber(overseasEntityDueDiligenceDto.getAmlNumber(), errors, loggingContext);
+        }
+        if (Objects.nonNull(overseasEntityDueDiligenceDto.getPartnerName())) {
+            validatePartnerName(overseasEntityDueDiligenceDto.getPartnerName(), errors, loggingContext);
+        }
+        return errors;
+    }
+
+    private boolean validateIdentityDate(LocalDate identityDate, Errors errors, String loggingContext) {
+        String qualifiedFieldName = getQualifiedFieldName(
+                OverseasEntitySubmissionDto.OVERSEAS_ENTITY_DUE_DILIGENCE,
+                OverseasEntityDueDiligenceDto.IDENTITY_DATE_FIELD);
+        return UtilsValidators.isNotNull(identityDate, qualifiedFieldName, errors, loggingContext) &&
+                DateValidators.isDateIsInPast(identityDate, qualifiedFieldName, errors, loggingContext) &&
+                DateValidators.isDateIsWithinLast3Months(identityDate, qualifiedFieldName, errors, loggingContext);
+    }
+
+    private boolean validateName(String name, Errors errors, String loggingContext) {
+        String qualifiedFieldName = getQualifiedFieldName(
+                OverseasEntitySubmissionDto.OVERSEAS_ENTITY_DUE_DILIGENCE,
+                OverseasEntityDueDiligenceDto.NAME_FIELD);
+        return StringValidators.isNotBlank(name, qualifiedFieldName, errors, loggingContext) &&
+               StringValidators.isLessThanOrEqualToMaxLength(name, MAX_LENGTH, qualifiedFieldName, errors, loggingContext) &&
```

Comment:
i see your point about the 256 , i assumed they all applied to same field. In which case, undo the MAX_LENGTH constant and pass the 256 as a param here (as original version). If we leave as is, it will confuse the reader as to why one is a constant and not others.

Reply:
Reversed the change.